rfcs/0002-ipfs-governance.md

```diff
@@ -0,0 +1,215 @@
+
+### The Core Working Group
+
+This whole structure and process relies on a Core Working Group, which acts in service of the other Teams and Working Groups to maintain a healthy, adaptable system for decision making and coordinated planning.
+
+The Core Working Group has three main functions:
+
+1. Set overall direction of the project
+2. Keep the Working Group lifecycle running smoothly
+3. Keep the RFC process running smoothly, delegating the decision making to Working Groups whenever possible
+
+### The Core Working Group is Responsible for
+- Delivering high-level, long-term strategic OKRs
+- Enforcing the lifecycle protocol for Teams and Working Groups: Guides formation of Teams/WGs, supports functioning of Teams & WGs, and dissolves them according to the lifecycle protocol
+- Keep the RFC process moving -- either delegating decision power to whichever Team/WG should make the decision or, when no other group matches the RFC, merging/rejecting the RFC themselves.
+
+### Prerogatives of the Core Working Group
+- Core team can assign/delegate ownership of a product, library, spec, problem domain to a Team or WG, So that those groups can operate with clear mandates and healthy amounts of autonomy.
+- In case of process failure or broken protocols among the teams, the Core Working Group is responsible for addressing the situation.
+- They have the authority to institute short-term measures to address pressing problems/disputes
+- They have the authority to fast-track RFCs that amend broken processes or protocols.
+
+### Membership in the Core Working Group
+
+The initial members of the Core Working Group are
+- Juan Benet (@jbenet) - BDFL
+- David Dias (@diasdavid) - Tech Lead for js-ipfs
+- Jeromy Johnson (@whyrusleeping) - Tech Lead for go-ipfs
+- Matt Zumwalt (@flyingzumwalt) - Program Manager for the IPFS ecosystem
+
+Future RFCs will define how membership of the Core Working Group will change over time. Note: For most open source orgs the "core team"/"core working group"/"steering group" is usually about 8-12 people.
+
+## Team and Working Group Lifecycle
+
+### 1. Design
+
+Anyone can propose creating a Team or Working Group. In order to be considered, it must satisfy a number of basic conditions:
+- A _Team_ **must have at least 3 people** whose **Full Time** efforts are primarily focused on that Team's work.
+- A _Working Group_ must be needed by at least 2 Teams or Working Groups.
+
+When preparing to propose a Team or Working Group, make sure to answer these questions:
+- What needs is the it addressing? Whose needs is it serving?
+- What does success look like? If this Team or WG achieves its goals, what will the results be?
+- Which other Teams and WGs will this group primarily interact with? Why? Have they expressed a need for it to exist?
+
+### 2. Proposal (RFC)
```

Comment:
I want to propose a `libp2p Team`, initially consisting of myself and Cole Brown.  I also want to propose a `libp2p Working Group` consisting of the team as well as others who are partially involved in libp2p development (like @vyzo or @Stebalien or @whyrusleeping or @diasdavid).

A couple problems I see:
1.  It seems weird that the libp2p team and WG's RFCs would be under the IPFS org, since we want libp2p to be an independent project.
2.  There are not technically two Teams requesting a libp2p Working Group.  I'm bending the definitions a little to allow libp2p to have a "Team" (in the sense of this RFC) and an "Extended Team" (a concept not addressed in this document but referring to a broader group of people with involvement in libp2p.

Thoughts on how I should approach this?  Maybe libp2p is exceptional because it is trying to become its own project at the same level as IPFS.  But I wonder how exceptional that really is.

Reply:
> There are not technically two Teams requesting a libp2p Working Group

It would not be difficult to extract a request for a libp2p WG from go-ipfs Team, Cluster team, libp2p team, js-ipfs team and maybe some other, if you ask nicely :)

Reply:
i think that's just premature; i don't think it's wise having two newbies in the project running on their own.
Let bigs get acquainted with the codebase first.